### PR TITLE
generalize validation to multi segments

### PR DESCRIPTION
**Pull request by EthanMarx**
Adds a `min_validation_duration` argument.

### CODE REVIEW

**Comment by EthanMarx** [comment]
@alecgunny @wbenoit26 Would be nice to get some quick feedback on this so I can re-launch the SV over time run. 
Adds minimum validation duration, and ability to run validation over multiple segments.

File: projects/sandbox/train/train/validation.py [hunk 1]
```diff
@@ -143,7 +144,9 @@ class Validator:
             checkpoints, and determines if training should be
             stopped
         background:
-            Tensor containing background data for each interferometer
+            Tensor containing background data for each interferometer.
+            Expected to be of shape
+            ∂`(num_segments, num_channels, num_samples)`.
```
Comment (wbenoit26) [inline projects/sandbox/train/train/validation.py:149]: I don't think this symbol is meant to be here

File: projects/sandbox/train/train/train.py [hunk 2]
```diff
@@ -241,9 +247,12 @@ def main(
     )
 
     if valid_waveforms is not None:
-        valid_fname = background_fnames.pop(-1)
-        logging.info(f"Loading validation segment {valid_fname}")
-        valid_background = train_utils.get_background(valid_fname)
+        logging.info(
+            "Loading validation segments "
+            f"{','.join(fname.stem for fname in valid_fnames)} "
```
Comment (wbenoit26) [inline projects/sandbox/train/train/train.py:252]: Looks like you need to make `fname` a `Path` here

File: projects/sandbox/train/tests/test_validation.py [hunk 3]
```diff
@@ -128,7 +129,7 @@ def test_shift_background(self, background):
 
     def test_iter_shift(self, background, psd_estimator, whitener):
         mock = Mock()
-        mock.background = background
+        mock.current_segment = background[0]
         mock.sample_rate = 8
```
Comment (wbenoit26) [inline projects/sandbox/train/tests/test_validation.py:133]: I think you need a `mock.current_duration = ` here and in `test_steps_for_shift`

**Pull request merged**